### PR TITLE
Generic types allowed in schema's extra types.

### PR DESCRIPTION
**Pull request by coady**
## Description
Generic types weren't resolving in the extra types of a schema.

## Types of Changes
- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #2154 

## Checklist
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Generic types allowed in schema's extra types.
```python
T = TypeVar('T')

@strawberry.type
class Node(Generic[T]):
    field: T

@strawberry.type
class Query:
    name: str

schema = strawberry.Schema(Query, types=[Node[int]])
```


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to A. Coady for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/schema/schema.py [hunk 1]
```diff
@@ -104,6 +105,9 @@ def __init__(
                     self.schema_converter.from_schema_directive(type_)
                 )
             else:
+                if hasattr(type_, "_type_definition"):
+                    if type_._type_definition.is_generic:
```
Comment (patrick91) [inline strawberry/schema/schema.py:109]: can we do this if in one line? :)

Comment (coady) [reply strawberry/schema/schema.py:109]: Black will reformat an `and` across 4 lines.

Comment (DoctorJohn) [reply strawberry/schema/schema.py:109]: I could live with four lines. To make the whole thing slightly prettier we could use a `continue` and remove the `else`. That would reduce the indentation level by one. Just an idea tho.

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Release type: patch
+
+ Generic types allowed in schema's extra types.
```
Comment (DoctorJohn) [inline RELEASE.md:3]: nit: will read better on Twitter :D
```suggestion
 Generic types are now allowed in the schema's extra types.
```

**Review by DoctorJohn — APPROVED** [review]
Overall looks good to me. Left two minor comments :)

File: strawberry/schema/schema.py (hunk 1 above)
Comment (DoctorJohn) [reply strawberry/schema/schema.py:109]: @patrick91 this PR overall looks good to me. Don't want to ignore your comment tho. Do you prefer 4 lines or 2 ifs? :)

**Review by DoctorJohn — APPROVED** [review]
Looks good to me now. Waiting for Patrick to answer on that one comment. Don't want to ignore his request :)

File: strawberry/schema/schema.py (hunk 1 above)
Comment (patrick91) [reply strawberry/schema/schema.py:109]: I'm fine as it is! Let's merge the PR 😊 Sorry @coady for the delay!

**Review by patrick91 — APPROVED** [review]

**Pull request merged**